### PR TITLE
Fix tests for GraalJS update

### PR DESCRIPTION
**Pull request by sdedic**
This is a followup to #7248 attempting to fix certain tests. IMHO it should be eventually merged into #7248 for overall testing before all the necessary changes go to `master`.

Keeping in Draft state now while adding more test fixes.

### CODE REVIEW

**Comment by sdedic** [comment]
For classloading in `core.network`: at some time GraalVM JDK included Javascript engine in it, it become available from the bootclasspath as well as Truffle libraries. The `libs.graalsdk.system` module addressed that by force-loading classes from the bootstrap. It is removed in @matthiasblaesing's GraalSDK update -- I need to think about that still a little, as there are older releases of GraalVM 11/17 that (still) include javascript engine. 
But because of potential clashes between JRE Truffle and NB-distributed Truffle/GraalJS, both libs.truffle.api and libs.graaljs explicitly hide truffle/js packages from the app classpath, so the NB modules can load their classes cleanly.

In the simplified test environment, all modules are loaded from the classpath; and the `OpenIDE-Hide-Classpath-Packages`  effectively hides module's own implementation from the tests classloader. I've added a `NbModuleSuite` wrapper around the tests -- but I think there's another option, to refine `OpenIDE-Hide-Classpath-Packages` to allow module-provided packages in the case the *module itself* is on the classpath. @jlahoda - have you an opinion on this ?

**Comment by matthiasblaesing** [comment]
Did you test this on GraalVM images? I tested with my branch and GraalVM 21 works like a normal VM, GraalVM 17 seems to be a problem.

File: java/nashorn.execution/test/unit/src/org/netbeans/modules/nashorn/execution/JSExecutorTest.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public void testLoadAllGraalJsLibraries() throws Exception {
         find.setAccessible(true);
         final ClassPath cp = (ClassPath) find.invoke(null);
         final FileObject[] roots = cp.getRoots();
-        assertEquals("Seven roots", 7, roots.length);
+        assertEquals("Seven roots", 11, roots.length);
```
Comment (mbien) [inline java/nashorn.execution/test/unit/src/org/netbeans/modules/nashorn/execution/JSExecutorTest.java:49]: nitpick: "11 roots expected"

File: webcommon/libs.graaljs/external/js-24.0.0-license.txt
Comment (mbien) [inline webcommon/libs.graaljs/external/js-24.0.0-license.txt:1]: i am no license expert so I can't comment on that, but i would likely have taken the GPL+CPE of the community edition

**Review by mbien — COMMENTED** [review]
test fixes make sense to me

**Pull request closed without merging**